functions/dependencies-notes.md

```diff
@@ -32,7 +45,13 @@ Without changing the code the best thing I've found to keep most of the dependen
 
 The best solution is to upgrade all of the firebase dependencies. Then we have to refactor the way the tests are working with the emulator and mocking the firebase admin object.
 
-The official firebase documentation recommends unit testing firebase using a real firebase project. This is fragile, so we'd prefer to keep using our emulator based approach.
+The official firebase documentation recommends unit testing firebase using a real firebase project or mocking things so it can run offline:
+https://firebase.google.com/docs/functions/unit-testing
+However, both options are fragile, so we'd prefer to keep using our emulator based approach.
+
+If we want to stick with using the rules unit testing library to implement our function tests against the emulator, there is this page which talks about unit tests with the emulator:
+https://firebase.google.com/docs/rules/unit-tests
+It refers to the v9 and v8 javascript SDK. It isn't clear what these versions are referring too.
```

Comment:
I would presume that this refers to the `firebase` SDK. We're still using v8 because v9 introduced substantial API changes. There's a `compat` mode in the v9 SDK designed to ease the transition but in a brief experiment a while ago I couldn't get it to work, but it would be worth trying again at some point.